### PR TITLE
[FSSDK-9871] Hook for track events

### PR DESCRIPTION
**Pull request by junaed-optimizely**
## Summary
- A functional approach to track events.
- `useTrackEvents` hook is exposed similar to the existing `useDecision` hook.

## Test plan
New test cases have been added to support the new functionality

## Issues
[FSSDK-9871](https://jira.sso.episerver.net/browse/FSSDK-9871)

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -17,7 +17,7 @@
 export { OptimizelyContext, OptimizelyContextConsumer, OptimizelyContextProvider } from './Context';
 export { OptimizelyProvider } from './Provider';
 export { OptimizelyFeature } from './Feature';
-export { useFeature, useExperiment, useDecision } from './hooks';
+export { useFeature, useExperiment, useDecision, useTrackEvents } from './hooks';
```
Comment (mikechu-optimizely) [inline src/index.ts:20]: NIT: I might follow the singular naming for your new hook, `useTrackEvent`

File: src/hooks.ts [hunk 2]
```diff
@@ -87,6 +87,10 @@ interface UseDecision {
   ];
 }
 
+interface UseTrackEvents {
```
Comment (mikechu-optimizely) [inline src/hooks.ts:90]: NIT: singular naming maybe?

```suggestion
interface UseTrackEvent {
```

File: README.md
Comment (mikechu-optimizely) [inline README.md:1]: 🔔 Reminder: update developer docs.

File: src/hooks.spec.tsx
Comment (mikechu-optimizely) [inline src/hooks.spec.tsx:1]: Please update each of these files' Copyright years.

**Review by mikechu-optimizely — CHANGES_REQUESTED** [review]
Great work. Thanks. I do not see any show-stoppers, but please review the following suggestions 👇

File: README.md
Comment (junaed-optimizely) [reply README.md:1]: Ok. Thats going to be an after-release task.

**Review by mikechu-optimizely — APPROVED** [review]
LGTM

**Pull request merged**